highway_env/envs/common/observation.py

```diff
@@ -158,7 +160,12 @@ def observe(self) -> np.ndarray:
         # Add ego-vehicle
         df = pd.DataFrame.from_records([self.env.vehicle.to_dict()])[self.features]
         # Add nearby traffic
-        sort, see_behind = (True, False) if self.order == "sorted" else (False, True)
+        if self.direction == "front":
```

Comment:
However, I think you misunderstood the roles of sort and see_behind, I do not think that the tree cases (front, behind, back) that you wrote correspond to the values you have set.
* see_behind is used to allow the observation of vehicles behind *and* in front the ego-vehicle (when true) vs only the vehicles in front (when false).
* sort will sort the observed vehicles by distance, which I don't think is related to your "front"/"behind"/"both" distinction

Reply:
As exemples : 

I run the code I wrote with "both", "front" and "behind" and I get these results:
With direction == "both":
```
[[ 1.          0.          0.25        0.36987187  0.        ]
 [ 1.          0.03682228  0.5         0.00789289  0.        ]
 [ 1.         -0.16690023  0.5        -0.02104962  0.        ]
 [ 1.          0.28638427  0.5         0.04154545  0.        ]
 [ 1.         -0.34303147  0.5        -0.01828814  0.        ]]
```
With direction == "front":
```
[[ 1.          0.          0.          0.36987187  0.        ]
 [ 1.          0.11172986  0.25        0.02657711  0.        ]
 [ 1.          0.28591441  0.75        0.02497863  0.        ]
 [ 1.          0.46706289  0.         -0.02210906  0.        ]
 [ 1.          0.60044589  0.          0.02134299  0.        ]]
```
With direction == "behind":
```
[[ 1.          0.          0.25        0.46346147  0.        ]
 [ 1.         -0.98732418  0.         -0.07317184  0.        ]
 [ 1.         -0.73644443  0.25       -0.05813579  0.        ]
 [ 1.         -0.60018745  0.         -0.07985231  0.        ]
 [ 1.         -0.33946373 -0.25       -0.07558893  0.        ]]
```
I understand that this behaviour is not what expected with the varibale `sort` and `see_behind`, so I think there is a problem with the function `self.env.road.close_vehicles_to()` because it's the behaviour of this function with these parameters.

Reply:
Ok I see in the function `self.env.road.close_vehicles_to()` your not taking the N nearest exo-vehicles but you're taking all the vehicles that are closer to the perception distance.
That makes that if you do not set `sort` to `True` the observation can show a list of exo-vehicles that are not the nearest, I think that's an error.
And that caused my misunderstanding